util/utils.go

```diff
@@ -32,7 +35,19 @@ func UrlClone(req *http.Request) *url.URL {
 		Path:       req.URL.Path,
 		RawPath:    req.URL.RawPath,
 		ForceQuery: req.URL.ForceQuery,
-		RawQuery:  	req.URL.RawQuery,
+		RawQuery:   req.URL.RawQuery,
 		Fragment:   req.URL.Fragment,
 	}
-}
\ No newline at end of file
+}
+
+func CreateRR(forwarder *forward.Forwarder, backends []Backend) *roundrobin.RoundRobin {
+	// randomize the list of backends to try to circumvent slightly biased load towards the beginning of the backend list (at high backend reconcile rates)
+	rand.Shuffle(len(backends), func(i, j int) { backends[i], backends[j] = backends[j], backends[i] })
```

Comment:
Pardon my Go ignorance, but don't you need to seed this first in order to get random results at each run?

Reply:
If `Shuffle` reseeds, I guess it's only deterministic for each lifetime of shelob? That's probably fine I guess.

Reply:
Thought I read somewhere that since Go something something rand is autoseeded, but... I guess it can't hurt to do it explicitly.

Reply:
Done! (94e45e2)